src/ucp/core/ucp_ep.c

```diff
@@ -357,12 +362,21 @@ ucs_status_t ucp_worker_create_mem_type_endpoints(ucp_worker_h worker)
 void ucp_worker_destroy_mem_type_endpoints(ucp_worker_h worker)
 {
     ucs_memory_type_t mem_type;
+    ucp_ep_h ep;
 
     ucs_memory_type_for_each(mem_type) {
-        if (worker->mem_type_ep[mem_type] != NULL) {
-           ucp_ep_destroy_internal(worker->mem_type_ep[mem_type]);
-           worker->mem_type_ep[mem_type] = NULL;
+        ep = worker->mem_type_ep[mem_type];
+        if (ep == NULL) {
+            continue;
         }
+
+        ucs_debug("memtype ep %p: destroy", ep);
+        ucs_assert(ep->flags & UCP_EP_FLAG_INTERNAL);
+        ucs_assert(ucp_ep_ext_control(ep)->local_ep_id == (ucs_ptr_map_key_t)ep);
```

Comment:
i think we can remove this assertion, ucp_ep_delete already checks it

Reply:
this assert is more strong
removed